web/components/groups/group-overview.tsx

```diff
@@ -0,0 +1,372 @@
+import { PostCard } from '../post-card'
+
+const MAX_TRENDING_POSTS = 6
+
+export function GroupOverview(props: {
+  group: Group
+  isEditable: boolean
+  posts: Post[]
+  aboutPost: Post | null
+  creator: User
+  user: User | null | undefined
+  memberIds: string[]
+}) {
+  const { group, isEditable, posts, aboutPost, creator, user, memberIds } =
+    props
+  return (
+    <Col className="pm:mx-10 gap-4 px-4 pb-12 pt-4 sm:pt-0">
+      <GroupOverviewPinned
+        group={group}
+        posts={posts}
+        isEditable={isEditable}
+      />
+
+      {(group.aboutPostId != null || isEditable) && (
+        <>
+          <SectionHeader label={'About'} href={'/post/' + group.slug} />
+          <GroupOverviewPost
+            group={group}
+            isEditable={isEditable}
+            post={aboutPost}
+          />
+        </>
+      )}
+      <SectionHeader label={'Trending'} />
+      <ContractSearch
+        user={user}
+        defaultSort={'score'}
+        noControls
+        maxResults={MAX_TRENDING_POSTS}
+        defaultFilter={'all'}
+        additionalFilter={{ groupSlug: group.slug }}
+        persistPrefix={`group-${group.slug}`}
+      />
+      <GroupAbout
+        group={group}
+        creator={creator}
+        isEditable={isEditable}
+        user={user}
+        memberIds={memberIds}
+      />
+    </Col>
+  )
+}
+
+function GroupOverviewPinned(props: {
+  group: Group
+  posts: Post[]
+  isEditable: boolean
+}) {
+  const { group, posts, isEditable } = props
+  const [pinned, setPinned] = useState<JSX.Element[]>([])
+  const [open, setOpen] = useState(false)
+  const [editMode, setEditMode] = useState(false)
+
+  useEffect(() => {
+    async function getPinned() {
+      if (group.pinnedItems == null) {
+        updateGroup(group, { pinnedItems: [] })
+      } else {
+        const itemComponents = await Promise.all(
+          group.pinnedItems.map(async (element) => {
+            if (element.type === 'post') {
+              const post = await getPost(element.itemId)
+              if (post) {
+                return <PostCard post={post as Post} />
+              }
+            } else if (element.type === 'contract') {
+              const contract = await getContractFromId(element.itemId)
+              if (contract) {
+                return <ContractCard contract={contract as Contract} />
+              }
+            }
+          })
+        )
+        setPinned(
+          itemComponents.filter(
+            (element) => element != undefined
+          ) as JSX.Element[]
+        )
+      }
+    }
+    getPinned()
+  }, [group, group.pinnedItems])
+
+  async function onSubmit(selectedItems: { itemId: string; type: string }[]) {
+    await updateGroup(group, {
+      pinnedItems: [
+        ...group.pinnedItems,
+        ...(selectedItems as { itemId: string; type: 'contract' | 'post' }[]),
+      ],
+    })
+    setOpen(false)
+  }
+
+  return isEditable || pinned.length > 0 ? (
```

Comment:
Would be nice to show a loading indicator if the pinned items aren't loaded, instead of "No pinned items yet".

I guess for non-editors, it's a better experience when no items were pinned to show no loading indicator...